### PR TITLE
Feature/synthetic crop data unity

### PR DESCRIPTION
**Pull request by nbarlowATI**
* Add function to generate random crop data on all shelves
* Another function to generate a random history of crop data for a given shelf
* API endpoints (to be called by unity app) to access above-mentioned synthetic data.

### CODE REVIEW

File: core/generate_synthetic_data.py [hunk 1]
```diff
@@ -265,3 +280,94 @@ def generate_weather_forecast():
...
+        waste_defect = 0
+        # some small chance of disease or defect (5% chance each)
+        r = random.random()
+        if r < 0.05:
+            # if there is disease, let it affect between 50 and 100% of crop
+            waste_disease = random.randint(5, 10) * 10
+        elif r < 0.1:
+            # same logic for defect
+            waste_defect = random.randint(5, 10) * 10
+        unit_yield = unit_yield * (1 - (waste_defect + waste_disease) / 100.0)
```
Comment (mhauru) [inline core/generate_synthetic_data.py:355]: This could go negative if we get unlucky.

File: webapp/app/queries/routes.py [hunk 2]
```diff
@@ -36,6 +36,10 @@
     jsonify_query_result,
     parse_date_range_argument,
 )
+from cropcore.generate_synthetic_data import (
+    generate_batches_in_farm,
```
Comment (mhauru) [inline webapp/app/queries/routes.py:40]: Does this get used somewhere?

**Review by mhauru — APPROVED** [review]
Looks good, I'm just confused about only one of the functions having an API end point I think.

File: webapp/app/queries/routes.py (hunk 2 above)
Comment (nbarlowATI) [reply webapp/app/queries/routes.py:40]: Eeek well spotted!! - that function+endpoint is the entire point of the PR (to get stuff into the unity demo), but somehow the endpoint function must have got lost when I was moving things around...   Restoring now.

**Pull request merged**